packages/typespec-ts/test/modularUnit/scenarios/models/serialization/enumKeyNorm.md

```diff
@@ -0,0 +1,145 @@
+# Should generate enum key normalization cases
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
+
+@service({
+  title: "Microsoft.Contoso management service",
+})
+@versioned(Microsoft.Contoso.Versions)
+namespace Microsoft.Contoso;
+
+union ExtensibleString {
+  pascal: "pascal",
+  pascalCase1: "pascalCase1",
+  PascalCase2: "PascalCase2",
+  pascalcase3: "pascalcase3",
+  Pascalcase4: "Pascalcase4",
+  pascal_case_5: "pascal_case_5",
+  pascal_case6: "pascal_case6",
+  _pascal_case7: "_pascal_case7",
+  `pascal, case8`: "pascal, case8",
+  MAX_of_MLD: "MAX_of_MLD",
+  // we will keep the upper case
+  YES_OR_NO1: "YES OR NO",
+  YES_OR_NO2: "YES OR NO",
+  VALIDATION_SUCCESS: "VALIDATION_SUCCESS",
+  ___pascal____case6666: "___pascal____case6666",
+  _10Pascal: "_10Pascal",
+  `090`: "090",
+  `10`: "10",
+  `20`: "20",
+  `1.0`: "1.0",
+  `-2.0`: "-2.0",
+  string,
+}
+
+union ExtensibleNumber {
+  One: 1,
+  `2`: 2,
+  `-2.1`: -2.1,
+  3,
+  int8,
+}
+
+
+enum Versions {
+  PreviewVersion: "2024-07-01-preview",
+  `2024-07-01`,
+  `2024-08-01-preview`
+}
+
+model Foo {
+  extensibleString: ExtensibleString;
+  extensibleNumber: ExtensibleNumber;
+}
+op post(@body body: Foo): void;
+@@clientName(ExtensibleString.`-2.0`, "$DO_NOT_NORMALIZE$Item-1.0");
+@@clientName(ExtensibleString.`YES_OR_NO2`, "Yes_Or_No2");
+// Issue here: https://github.com/microsoft/typespec/issues/5081
+// @@clientName(ExtensibleNumber.3, "Enum3");
+// TODO: renaming is not working in compiler v0.61 but should be fixed in v0.62
```

Comment:
Please note enaming is not working in compiler v0.61 but should be fixed in v0.62 for version enum.